addOns/ascanrulesAlpha/src/main/java/org/zaproxy/zap/extension/ascanrulesAlpha/CorsScanRule.java

```diff
@@ -0,0 +1,189 @@
+                    risk = Alert.RISK_MEDIUM;
+                } else if (acaoVal.contains(RANDOM_NAME)
+                        || acaoVal.contains("null")
+                        || (secScheme && acaoVal.contains("http:"))) {
+                    vuln = true;
+                    // If authenticated AJAX requests are allowed, the risk is higher
+                    if (acacVal == "") risk = Alert.RISK_MEDIUM;
+                    else risk = Alert.RISK_HIGH;
+                }
+                Matcher m =
+                        Pattern.compile(
+                                        String.format(
+                                                "^\\s*%s[:\\s]+%s(\\s+%s[:\\s]+%s)?",
+                                                HttpHeader.ACCESS_CONTROL_ALLOW_ORIGIN,
+                                                Pattern.quote(acaoVal),
+                                                ACAC,
+                                                acacVal),
+                                        Pattern.MULTILINE)
+                                .matcher(respHead.toString());
+                buildAlert(risk)
+                        .setMessage(msg)
+                        .setAttack(HttpRequestHeader.ORIGIN + ": " + payload)
+                        .setEvidence(m.find() ? m.group(0) : null)
+                        .raise();
+                return;
+            } catch (IOException e) {
+                LOG.error(e.getMessage(), e);
+            }
+        }
+    }
+
+    private AlertBuilder buildAlert(int risk) {
+
+        return newAlert()
+                .setName(risk == Alert.RISK_INFO ? getName() : getConstantStr("vuln.name"))
+                .setRisk(risk)
+                .setConfidence(Alert.CONFIDENCE_HIGH)
+                .setDescription(
+                        risk == Alert.RISK_INFO ? getDescription() : getConstantStr("vuln.desc"));
+    }
+
+    private static String getConstantStr(String suffix) {
+        return Constant.messages.getString("ascanalpha.cors." + suffix);
+    }
+
+    @Override
+    public int getId() {
+        return 40040;
+    }
+
+    @Override
+    public String getName() {
+        return getConstantStr("info.name");
+    }
+
+    @Override
+    public String getDescription() {
+        return getConstantStr("info.desc");
+    }
+
+    @Override
+    public int getCategory() {
+        return Category.SERVER;
+    }
+
+    @Override
+    public String getSolution() {
+        return getConstantStr("soln");
+    }
+
+    @Override
+    public String getReference() {
+        return getConstantStr("refs");
+    }
+
+    @Override
+    public int getCweId() {
+        return 942;
+    }
+
+    @Override
+    public int getWascId() {
+        return 14;
+    }
+
+    public List<Alert> getExampleAlerts() {
+        List<Alert> alerts = new ArrayList<Alert>();
+        for (int i = Alert.RISK_INFO; i < Alert.RISK_HIGH; i++) alerts.add(buildAlert(i).build());
+        return alerts;
+    }
```

Comment:
This should return actual relevant examples for this scan rule.  There are names and descriptions relevant to only certain risk levels, Along with relevant URL, attack, param, evidence, etc.

Have a look through other scan rules that implement the functionality for ideas.

Reply:
I think this function is what you want. It calls buildAlert with the possible risks levels (Info, Medium, High) and buildAlert will return alerts with different names and descriptions depending of the risk level. Do I miss something?

Reply:
No, I may not have read through it closely enough. I’ll pull it later and do some testing.